### DISCUSSION TITLE
Issues with Annotator

### DISCUSSION THREAD

**Discussion by jcesguerram** [RyotaUshio/obsidian-pdf-plus#94] [Q&A]
Ryota, can you explain a little more what you mean when you say that Annotator (or something similar) would "leave behind a pile of unreadable JSON" if it stops working?
does that mean that the annotations made in pdfs would no longer be visible?

**Comment by RyotaUshio** [comment]
You can see what I mean by temporarily disabling Annotator.

So let's say we have a PDF annotated like this:

<img width="1552" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/60cb92af-4b72-41f9-b603-5880f33ff0cc">

This annotation is in fact stored as a JSON code block in the markdown file you added the `annotation-target` property:

<img width="1552" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/069b7e92-a8b1-4057-9ed5-955dc19b811c">

Annotator reads the annotation data from this JSON code block and **just displays** it in their PDF viewer.

---

Now suppose Annotator has stopped working. 
Then, your annotation is still stored in the markdown file as a JSON code block, but the md file is no longer shown as a PDF viewer, and you can't see it as a highlight in the PDF anymore. 
Also note that Annotator does not support exporting highlights (https://github.com/elias-sundqvist/obsidian-annotator/issues/110).

Of course, your annotation data is not totally lost because it's still stored in the markdown file as a JSON code block. But it's not written for humans so it's hard to interpret.

**Reply by RyotaUshio** [reply]
One thing to note is that highlights added by PDF++'s backlink highlighting feature will also be invisible if PDF++ dies.
But PDF++ is advantageous in that

- you can choose to add highlights directly into PDF files so that they are visible even after this plugin stops working.
- PDF++ uses Obsidian's built-in link syntax (`#page=...&selection=...`) to express what part of the text to highlight. These links will keep functioning even if PDF++ is gone (because it's built-in).
- Your comments are pure markdown (unlike Annotator's original JSON format) so they can keep being parts of your vault as before
- I'm planning to add an exporting feature for backlink highlights!

**Reply by jcesguerram** [reply]
Thanks for the replies!
One question: One cool feature of Annotator is that if you click in one of the comments/highlights in the annotation pane it takes you to taht part of the pdf document. could that be done in your plugin (by clicking in the highlights in the backlink section)?

**Reply by jcesguerram** [reply]
@RyotaUshio like in the first image ypu shared in this post. in annotatorm if you click on annotation in the right, it will take ypu to that part of the pdf document. would it be possible to do that in Pdf+?

**Reply by RyotaUshio** [reply]
sorry for the late reply, currently PDF++ doesn't support exactly the same functionality but instead it requires two steps:

- click a backlink pane and open the markdown file that contains the link to the PDF
- in the markdown file, click the PDF link to move to the corresponding location in the PDF

The feature you described will be a great addition, but it will require creating PDF++'s own version of the backlink view or comments pane, so it will not happen in the short term. But I agree it will be helpful and I'd like to add it somewhen in the future.

**Reply by jcesguerram** [reply]
thank you su much for the reply and for all the work you do. i'm going to 'buy you a coffee'

**Reply by RyotaUshio** [reply]
Wow that's wonderful, thank you so much!
